site/packages/aa-signers/turnkey/authenticate.md

```diff
@@ -0,0 +1,79 @@
+---
+outline: deep
+head:
+  - - meta
+    - property: og:title
+      content: TurnkeySigner • authenticate
+  - - meta
+    - name: description
+      content: Overview of the authenticate method on TurnkeySigner
+  - - meta
+    - property: og:description
+      content: Overview of the authenticate method on TurnkeySigner
+---
+
+# authenticate
+
+`authenticate` is a method on the `TurnkeySigner` which leverages the `Turnkey` SDK to authenticate a user.
+
+This method must be called before accessing the other methods available on the `TurnkeySigner`, such as signing messages or typed data or accessing user details.
+
+## Usage
+
+::: code-group
+
+```ts [example.ts]
+// [!code focus:99]
+import { TurnkeySigner } from "@alchemy/aa-signers";
+import { WebauthnStamper } from "@turnkey/webauthn-stamper";
+import { http } from "viem";
+
+// Create the Turnkey client
+const turnkeyClient = new TurnkeyClient(
+  {
+    baseUrl: "api.turnkey.com",
+  },
+  new WebauthnStamper({
+    rpId: "your.app.xyz",
+  }),
+);
+
+// If you need to retrieve your user's wallet ID from Turnkey
+const wallets = await turnkeyClient.getWallets({
+  organizationId: "TURNKEY_ORGANIZATION_ID",
+});
+if (wallets.length === 0) throw new Error("No wallets created!");
```

Comment:
for a new user this will always be true right? maybe let's add a call to `createWallet` or `createSubOrg` here so we can generate a new user / wallet

Reply:
by here you mean as an object on `authParams` right? agree, would love this to be part of the `authenticate` flow.

Reply:
this is a great call out @avasisht23 ! i wrote the code with the assumption that between initializing `turnkeyClient` and `turnkeySigner`, the dev creates or retrieves a sub-organization and wallet for the user. but as @moldy530 refers to in the first comment, there's multiple flows, including:
- already have sub-organization id and wallet -> `getWallets`/`getWallet`
- already have sub-organization -> `createWallet`
- `createSubOrganization`

to capture this flow as a method in `authParams` that the dev can implement, i've replaced `organizationId` and `signWith` with `resolveSubOrganization()`. i'm going to be honest -- the turnkey sdk could really be improved to simplify the dev's current implementation. fortunately, we're working on some design changes to the turnkey sdk. for example, these `create` endpoints will become synchronous and their new response type can be used instead of maintaining a `TurnkeySubOrganization` type in turnkey/types. (related to: https://github.com/alchemyplatform/aa-sdk/pull/292#discussion_r1414216980)